packages/date-picker/src/types.ts

```diff
@@ -66,3 +66,27 @@ export interface DisabledDayMatchers {
    */
   disabledAfter?: Date
 }
+
+export type DateRangeValidationStatus = {
+  dateStart?: FieldMessageStatus
+  dateEnd?: FieldMessageStatus
+}
+
+export type DateRangeMessageType = {
+  dateStart: React.ReactNode
+  dateEnd: React.ReactNode
+}
+
+export type DateRangeValidationMessageType =
+  | React.ReactNode
+  | DateRangeMessageType
+
+export function isDateRangeMessageType(
+  message: DateRangeValidationMessageType
+): message is DateRangeMessageType {
+  return (
+    message != undefined &&
+    (message as DateRangeMessageType).dateStart !== undefined &&
```

Comment:
Shouldn't it be `or`? You can probably write a unit test for this to confirm it's doing what you're expecting 😉 

```suggestion
    (message as DateRangeMessageType).dateStart !== undefined ||
```

Reply:
How I thought the API would be that the consumer either passes a React.ReactNode or the 
```
{
  dateStart: React.ReactNode
  dateEnd: React.ReactNode
}
```

where both `dateStart` and `dateEnd` are required. Otherwise if they only wanted one then they can pass it via the `React.ReactNode`?